packages/@dcl/ecs/test/components/OnPointerDown.spec.ts

```diff
@@ -0,0 +1,64 @@
+﻿import { Engine } from '../../src/engine'
+
+describe('Generated OnPointerDown ProtoBuf', () => {
+  it('should serialize/deserialize OnPointerDown', () => {
+    const newEngine = Engine()
+    const { OnPointerDown } = newEngine.baseComponents
+    const entity = newEngine.addEntity()
+    const entityB = newEngine.addEntity()
+    OnPointerDown.create(newEngine.addEntity())
+    const onPointerDown = OnPointerDown.create(entity, {
+      identifier: '98352439-fdsfds',
+      button: 'PRIMARY',
+      hoverText: 'Tap to run',
+      distance: 10,
+      showFeedback: true
+    })
+
+    OnPointerDown.create(entityB, {
+      identifier: 'fg98352439-fdsfds',
+      button: 'D',
+      hoverText: 'TCker',
+      distance: 5,
+      showFeedback: false
+    })
+    const buffer = OnPointerDown.toBinary(entity)
+    OnPointerDown.updateFromBinary(entityB, buffer)
+
+    expect(onPointerDown).toEqual({ ...OnPointerDown.mutable(entityB) })
+  })
+
+  it('should receive OnPointerResult', () => {
+    const newEngine = Engine()
+    const { OnPointerDown, OnPointerResult } = newEngine.baseComponents
+    const entity = newEngine.addEntity()
+    OnPointerDown.create(newEngine.addEntity())
+
+    // We create an onPointerDownEvent
+    const onPointerDown = OnPointerDown.create(entity, {
+      identifier: '98352439-fdsfds',
+      button: 'PRIMARY',
+      hoverText: 'Tap to run',
+      distance: 10,
+      showFeedback: true
+    })
+
+    // wait a tick to receive a response
+    newEngine.update(1 / 30)
+
+    // We receive an OnPointerResult
+    const onPointerResult = OnPointerResult.create(entity, {
+      identifier: '98352439-fdsfds',
+      button: 'PRIMARY',
```

Comment:
Same here, an int or enum would be faster and cheaper for the serialization

Reply:
ANY